include/boost/integer/extended_euclidean.hpp

```diff
@@ -0,0 +1,73 @@
+/*
+ *  (C) Copyright Nick Thompson 2018.
+ *  Use, modification and distribution are subject to the
+ *  Boost Software License, Version 1.0. (See accompanying file
+ *  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+ */
+#ifndef BOOST_INTEGER_EXTENDED_EUCLIDEAN_HPP
+#define BOOST_INTEGER_EXTENDED_EUCLIDEAN_HPP
+#include <limits>
+#include <stdexcept>
+#include <boost/throw_exception.hpp>
+
+namespace boost { namespace integer {
+
+// From "The Joy of Factoring", Algorithm 2.7, with a small optimization to remove tmps from Wikipedia.
+// Solves mx + ny = gcd(m,n). Returns tuple with (gcd(m,n), x, y).
+
+template<class Z>
+struct euclidean_result_t {
+  Z gcd;
+  Z x;
+  Z y;
+};
+
+using std::numeric_limits;
+
+template<class Z>
+euclidean_result_t<typename std::enable_if<numeric_limits< Z >::is_signed, Z>::type>
+extended_euclidean(Z m, Z n)
+{
+    if (m < 1 || n < 1)
+    {
+        BOOST_THROW_EXCEPTION(std::domain_error("Arguments must be strictly positive.\n"));
+    }
+
+    bool swapped = false;
+    if (m < n)
+    {
+      swapped = true;
+      std::swap(m, n);
```

Comment:
Change this to `using std::swap` and an unqualified call to `swap` so that any user-defined overloads are picked up.

Reply:
Yup, good point. Done.